packages/api/lambdas/kinesis-consumer.js

```diff
@@ -34,63 +36,74 @@ async function getKinesisRules(event) {
 }
 
 /**
- * `createOneTimeRules` creates new rules with the same data as a kinesis-type rule, except the type is modified to 'onetime'.
+ * Queue a workflow message for the kinesis rule with the message passed
+ * to kinesis as the payload
  *
- * @param {array} kinesisRules list of rule objects
- * @returns {array} Array of promises for model.create
+ * @param {Object} kinesisRule - kinesis rule to queue the message for
+ * @param {Object} eventObject - message passed to kinesis
+ * @returns {Promise} promise resolved when the message is queued
  */
-async function createOneTimeRules(kinesisRules) {
-  const oneTimeRulePromises = kinesisRules.map((kinesisRule) => {
-    const oneTimeRuleParams = Object.assign({}, kinesisRule);
-    delete oneTimeRuleParams['createdAt'];
-    delete oneTimeRuleParams['updatedAt'];
-    oneTimeRuleParams.name = `${kinesisRule.name}_${Date.now().toString()}`;
-    oneTimeRuleParams.rule.type = 'onetime';
-    return model.create(oneTimeRuleParams);
-  });
+async function queueMessageForRule(kinesisRule, eventObject) {
+  const item = {
+    workflow: kinesisRule.workflow,
+    provider: kinesisRule.provider,
+    collection: kinesisRule.collection,
+    payload: eventObject
+  };
 
-  return await Promise.all(oneTimeRulePromises);
+  return Rule.buildPayload(item)
+    .then((payload) => queueWorkflowMessage(payload));
```

Comment:
Sorry missed this the first time around, this could be:
```javascript
return Rule.buildPayload(item).then(queueWorkflowMessage);
```

Reply:
Nice, thanks